src/pivot_comb.jl

```diff
@@ -93,5 +102,8 @@ function pivot_comb(A,
     end
 
     X[(!).(P)] .= 0.0
+    if return_P
+        return X, P
```

Comment:
this type of "if" leads to undesirable type instability.
it would be better to have a separate function that always returns both outputs.
then this function can just return the 1st output of that separate function.